src/Promise.php

```diff
@@ -10,14 +10,9 @@
     /**
      * Registers a callback to be invoked when the promise is resolved.
      *
-     * The callback receives `null` as first parameter and `$value` as second parameter on success. It receives the
-     * failure reason as first parameter and `null` as second parameter on failure.
-     *
      * If the promise is already resolved, the callback MUST be executed immediately.
      *
-     * Warning: If you use type declarations for `$value`, be sure to make them accept `null` in case of failures.
```

Comment:
The name of parameter is optional, and superfluous when there is only one parameter.

Given a proper description of the callable, which now exists, I can't see a reason to have further warnings, it is strongly implied, if not very obvious given the good description.